### PR TITLE
ENT-1181 | Updating piptools for OEP-18

### PR DESCRIPTION
**Pull request by christopappas**
- Adds the files/script necessary for compiling `.in` into `.txt`
- Adds `make upgrade` functionality to Makefile
- Replaces `django_sites_extensions.middleware.CurrentSiteWithDefaultMiddleware`  with `django.contrib.sites.middleware.CurrentSiteMiddleware`. Why? The `edx-django-sites-extensions` version was very old and was incompatible with Django 1.11. I originally upgraded `edx-django-sites-extensions` to its latest version 2.4.0, but this version did not include `CurrentSiteWithDefaultMiddleware`. After completely removing the package and replacing its middleware with the the Django CurrentSiteMiddleware, all tests still passed.
- ~Comments out `pyinotify` from `requirements/dev.in` as it breaks Mac OSX installations.~

### CODE REVIEW

File: requirements/dev.in [hunk 1]
```diff
@@ -12,4 +11,4 @@ edx-i18n-tools==0.4.4
 ptvsd==3.0
 
 # For devserver code reloading
-pyinotify==0.9.6
+#pyinotify==0.9.6 # This breaks OSX installation
```
Comment (georgebabey) [inline requirements/dev.in]: As we talked about, most people are running ecomm through devstack which runs in linux, not on their local macs. Given that this should probably be committed uncommented and if you find you need to use a virtual env on your mac you can always comment it out.

**Comment by georgebabey** [comment]
Regarding the `CurrentSiteMiddleware` I believe you are correct as the README for [edx-django-sites-extensions](https://github.com/edx/edx-django-sites-extensions) mentions the package you are including.  Looks like this PR was the one that made the change by introducing our own redirect middleware: https://github.com/edx/edx-django-sites-extensions/pull/6

**Comment by georgebabey** [comment]
I noticed we're pinning all of the version in the .in files. I believe the goal of this was the leave these unpinned when possible, so that someone running `make upgrade`  That said, given how long they've been pinned, I'm sure many are out of date.

@robrap Any thoughts on how you'd like to see this move forward?  We could merge with the pinned versions and gradually unpin them or we could take a stab at unpinning them here, running `make upgrade and seeing what breaks...

**Comment by robrap** [comment]
@georgebabey: I'd say some combo.  Keep them pinned to land this quicker.  Let's have a follow-up to see what's the best way to get as much unpinned as possible as fast as possible.  Your team can try to do some of that if that is simple as part of this ticket, but a separate PR would probably make your life easier.

**Review by georgebabey — APPROVED** [review]

**Comment by georgebabey** [comment]
@christopappas Can you rebase your branch with master so we can merge this in?

**Pull request merged**